song-server/src/main/java/bio/overture/song/server/controller/SchemaController.java

```diff
@@ -16,39 +16,64 @@
  */
 package bio.overture.song.server.controller;
 
-import bio.overture.song.server.model.dto.schema.GetSchemaResponse;
-import bio.overture.song.server.model.dto.schema.ListSchemaIdsResponse;
-import bio.overture.song.server.service.SchemaService;
+import bio.overture.song.server.model.dto.AnalysisType;
+import bio.overture.song.server.service.AnalysisTypeService;
 import io.swagger.annotations.Api;
 import io.swagger.annotations.ApiOperation;
-import lombok.RequiredArgsConstructor;
+import lombok.NonNull;
+import lombok.SneakyThrows;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.Collection;
+
+import static bio.overture.song.core.utils.JsonUtils.readTree;
+import static bio.overture.song.core.utils.JsonUtils.toPrettyJson;
+
 @RestController
-@RequestMapping(path = "/schema")
-@RequiredArgsConstructor
+@RequestMapping(path = "/schemas")
 @Api(tags = "Schema", description = "Get schemas used for uploads")
 public class SchemaController {
 
+  private final AnalysisTypeService analysisTypeService;
+
   @Autowired
-  private SchemaService schemaService;
+  public SchemaController(@NonNull AnalysisTypeService analysisTypeService) {
+    this.analysisTypeService = analysisTypeService;
+  }
+
+  @ApiOperation(value = "ListAnalysisTypes",
+      notes = "Retrieves a list of registered analysisType names" )
+  @GetMapping("/analysis")
+  public Collection<String> listAnalysisTypes(){
+    return analysisTypeService.listAnalysisTypeNames();
+  }
+
+  @ApiOperation(value = "GetAnalysisTypeVersion",
+      notes = "Retrieves a specific version of a schema for an analysisType" )
+  @GetMapping("/analysis/{name}/{version}")
+  public AnalysisType getAnalysisTypeVersion(
+      @PathVariable("name") String name,
+      @PathVariable("version") Integer version){
+    return analysisTypeService.getAnalysisType(name, version);
+  }
 
-  @ApiOperation(value = "ListSchemasIds",
-      notes = "Retrieves a list of registered schema ids" )
-  @GetMapping("/list")
-  public ListSchemaIdsResponse listSchemaIds(){
-    return schemaService.listSchemaIds();
+  @ApiOperation(value = "GetLatestAnalysisType",
+      notes = "Retrieves the latest version of a schema for an analysisType" )
+  @GetMapping("/analysis/{name}/latest")
```

Comment:
This should also follow the same `GET /schemas?name=<name>&order=<order>&limit=1` or something like that. 
The elements of the path in a RESTful design should be nouns.